stlab/concurrency/future.hpp

```diff
@@ -296,20 +297,20 @@ struct shared_base<T, enable_if_copyable<T>> : std::enable_shared_from_this<shar
     template <typename F, typename... Args>
     void set_value(F& f, Args&&... args);
 
-    auto get_try() -> boost::optional<T> {
+    auto get_try() -> boost::optional<const T&> {
         bool ready = false;
         {
             std::unique_lock<std::mutex> lock(_mutex);
             ready = _ready;
         }
         if (ready) {
             if (_error) std::rethrow_exception(_error.get());
-            return _result;
+            return _result.value();
         }
         return boost::none;
     }
 
-    auto get_try_r(bool unique) -> boost::optional<T> {
+    auto get_try_r(bool unique) -> boost::optional<const T&> {
```

Comment:
This seems very dangerous - if this is an rvalue reference then the value being referenced is about to go away... It also potentially forces the client to do a copy when a move would have sufficed.